pandas/core/internals/blocks.py

```diff
@@ -2341,30 +2341,26 @@ def get_block_type(dtype: DtypeObj):
     -------
     cls : class, subclass of Block
     """
-    # We use kind checks because it is much more performant
-    #  than is_foo_dtype
-    kind = dtype.kind
-
-    cls: type[Block]
-
     if isinstance(dtype, SparseDtype):
```

Comment:
i think the SparseDtype check may no longer be needed

Reply:
removed

Reply:
your suggested updates give a bit of an improvement to non-EA's as well:

```
import numpy as np
from pandas.core.internals.blocks import get_block_type

%timeit get_block_type(np.dtype('float64'))

# 724 ns ± 59.4 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)  -> main
# 590 ns ± 30 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)    -> PR